f8a_report/sentry_report_helper.py

```diff
@@ -0,0 +1,132 @@
+            events = self.retrieve_events(item['id'])
+            errors['id'] = item['id']
+            errors['lastSeen'] = item['lastSeen']
+            errors[events['pods_impacted']] = item['metadata']['type'] + ": " + \
+                item['metadata']['value'] if item['metadata'].get('type')\
+                else item['metadata']['title']
+            errors['stacktrace'] = events['stacktrace']
+            # Detecting the endpoint services
+            endpointpodname = events['pods_impacted'].split('-')
+            endpointpodname.pop(len(endpointpodname) - 1)
+            endpointpodname.pop(len(endpointpodname) - 1)
+            server_name = "-".join(endpointpodname)
+            result['error_report'][server_name] = result['error_report'][server_name] \
+                if result['error_report'].get(server_name) else {}
+            # Calculating total errors
+            result['error_report'][server_name]['total_errors'] = \
+                result['error_report'][server_name]['total_errors'] + 1 \
+                if result['error_report'][server_name].get('total_errors') else 1
+            if not result['error_report'][server_name].get('errors'):
+                result['error_report'][server_name]['errors'] = []
+            result['error_report'][server_name]['errors'].append(errors)
+            # Saving the final report in the relevant S3 bucket
+
+            try:
+                obj_key = '{depl_prefix}/{type}/{report_name}.json'.format(
+                    depl_prefix=self.s3.deployment_prefix, type=report_type, report_name=report_name
+                )
+                self.s3.store_json_content(content=result, obj_key=obj_key,
+                                           bucket_name=self.s3.report_bucket_name)
+            except Exception as e:
+                logger.exception('Unable to store the report on S3. Reason: %r' % e)
+
+        return result
+
+    def retrieve_events(self, issue_id):
+        """Retrieve results for issue events."""
+        events = {'stacktrace': ''}
+
+        try:
+            # Invoke Sentry API to run the event collection
+            auth = 'Bearer {token}'.format(token=self.sentry_token)
+            resp = requests.get(url=self.sentry_api_tags + issue_id + '/events/latest/',
+                                headers={"Authorization": auth})
+            # Check for status code
+            # If status is not success, log it as an error
+            if resp.status_code == 200:
+                logger.info('Successfully invoked Sentry API \n {resp}'.format(resp=resp.json()))
+                output = resp.json()
+                # retrieving server name info
+                for item in output['tags']:
+                    if item['key'] == 'server_name':
+                        events['pods_impacted'] = item['value']
+                        break
+                # Collecting stacktrace for each frames
+                if output['entries'][1]['type'] == 'exception':
```

Comment:
What if the length of the output['entries] is less than 1? Does this hardcoding works all the time, or its better to check the length

Reply:
@miteshvp I have seen at-least couple of error stacks for this. Usually, output['entries] has 3-4 components: message, exception, breadcrumb, request where in some cases exception component may be missing where stacktrace is not available. So, length will be greater than 1. I think, length check is not required.

Reply:
@animuk let's either go by Mitesh's approach of checking length or handle exception for IndexError and KeyError here.

Reply:
handled exception for IndexError and KeyError here.